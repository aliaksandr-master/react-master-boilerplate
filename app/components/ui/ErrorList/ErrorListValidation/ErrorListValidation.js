import { compose, setDisplayName } from 'recompose';
import ErrorListValidationView from './ErrorListValidation.view';



export default compose(
  setDisplayName('ErrorListValidation')
)(ErrorListValidationView);
