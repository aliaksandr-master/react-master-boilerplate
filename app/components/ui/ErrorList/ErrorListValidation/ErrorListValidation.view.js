import React from 'react';
import PropTypes from 'prop-types';
import capitalize from 'lodash/capitalize';
import classnames from '../../../../vendor/classnames';
import errorPropType from '../../../../prop-types/error';
import './ErrorListValidation.view.less';



const ErrorListValidation = ({ error: { message, typed_params: { field_name } }, fieldTitleMap, className }) => {
  const displayFieldName = fieldTitleMap && fieldTitleMap.hasOwnProperty(field_name) ? (fieldTitleMap[field_name] || '') : capitalize(String(field_name).replace('_', ' '));

  return (<div className={classnames('ErrorListValidation', className)}>{`${displayFieldName} : ${message}`}</div>);
};



ErrorListValidation.propTypes = {
  className: PropTypes.string,
  error: errorPropType.isRequired,
  fieldTitleMap: PropTypes.objectOf(PropTypes.string)
};



ErrorListValidation.defaultProps = {
};



export default ErrorListValidation;
