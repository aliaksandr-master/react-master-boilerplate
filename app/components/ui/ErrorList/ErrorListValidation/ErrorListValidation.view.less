@import '../../../../lib/mixins.less';

.ErrorListValidation {
  margin-top: @pad*2;

  &:first-child {
    margin-top: 0;
  }
}
