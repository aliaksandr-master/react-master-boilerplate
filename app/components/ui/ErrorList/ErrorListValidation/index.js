import ErrorListValidation from './ErrorListValidation';



export default ErrorListValidation;
