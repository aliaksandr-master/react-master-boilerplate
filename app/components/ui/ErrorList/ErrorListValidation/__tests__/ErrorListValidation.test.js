/* eslint-env browser, jest */
/* eslint-disable react/jsx-no-bind */

import React from 'react';
import renderer from 'react-test-renderer';
import ErrorListValidation from '../index';
import Test from '../../../../test/Test';



describe('rendering', () => {
  it('should render something', () => {
    const component = renderer.create(
      <Test>
        <ErrorListValidation
          error={{
            message: 'ipsum',
            typed_params: {
              field_name: 'lorem'
            }
          }}
          fieldTitleMap={{}}
        />
      </Test>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
