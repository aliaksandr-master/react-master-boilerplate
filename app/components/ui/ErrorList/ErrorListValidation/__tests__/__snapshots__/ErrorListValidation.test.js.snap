// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render something 1`] = `
<div
  className="ErrorListValidation"
>
  Lorem : ipsum
</div>
`;
